taskcluster/mozillavpn_taskgraph/transforms/release_index.py

```diff
@@ -0,0 +1,23 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+
+from taskgraph.transforms.base import TransformSequence
+transforms = TransformSequence()
+
+
+@transforms.add
+def add_pr_number(config, tasks):
+   for task in tasks:
+        if task.__contains__("push-to-release-index"):
+            release_index = task.pop("push-to-release-index")
+            if release_index and int(config.params["level"]) == 3:
```

Comment:
I think we'll want to check `config.params["tasks_for"] == "github-release"` here. Otherwise, any old push to a branch will overwrite this index.

Reply:
I think that is what we want rn. Even as we are in "feature freeze" we will update the release branch probably multiple times per day, and we want to have the index to always point at the latest build here. 
Doing a gh-pre-release for that would not really scale here and make qa depend on us pushing that often :D 
Also once we are in codefreeze main will host 2.10, so i'm not worried about main or other branches overwriting this.

Reply:
Thinking about... maybe in that case the index should not be `mozillavpn.releases.{version}.{name}` rather `mozillavpn.preview.{version}.{name}` ? And keep the mozillavpn.releases for "github-release" graphs?

Reply:
Yeah, I think that's what I was missing.

So in this case I'd recommend using the `branch` namespace, e.g:
```
index.mozillvpn.v2.mozilla-vpn-client.branch.{branch}.latest.{name}
```
We already index the decision task there, so would just be a matter of indexing the builds as well:
https://firefox-ci-tc.services.mozilla.com/tasks/index/mozillavpn.v2.mozilla-vpn-client.branch.main

The good news (or maybe bad if you spent a lot of time on this) is that the `generic` index builder should do this for us already. So you can likely just add the `index` key to the signing tasks and it might work.

Reply:
My bad, the `generic` index build does not add these `branch` routes.

So your patch is still necessary, but I'd suggest copying something similar to the above route. We'll likely want to use the unsigned builds here as well as I think we may stop doing build signing on every push eventually.